app/uk/gov/hmrc/gatekeeper/views/emails/EmailAllUsersNewView.scala.html

```diff
@@ -24,20 +24,101 @@
 
 @this(main: MainView, applicationConfig: AppConfig)
 
-@(users: Seq[RegisteredUser], composeEmailRecipients: JsValue, emails: String)(implicit request: Request[_], loggedInUser: LoggedInUser, messagesProvider: MessagesProvider)
-
-@main(
-    title = s"${applicationConfig.title} - Emails all users",
-    breadcrumbs = Seq(
-        Crumb(
-            "Back",
-            s"${routes.EmailsPreferencesController.showEmailInformation("all-users")}",
-            Some("data-back-link")
+@(users: Seq[RegisteredUser],
+        composeEmailRecipients: JsValue,
+        emails: String,
+        offset: Int,
+        limit: Int,
+        totalCount: Int)(implicit request: Request[_], loggedInUser: LoggedInUser, messagesProvider: MessagesProvider)
+
+    @currentPage = @{
+        (offset / limit) + 1
+    }
+
+    @lastPage = @{
+        val lp = (totalCount / limit)
+        val remainder =  (totalCount % limit)
+        if(remainder < limit) lp + 1 else remainder / limit
+    }
```

Comment:
```suggestion
    @lastPage = @{
        ceil(totalCount.toDouble / limit).toInt
    }
```
and also @import `scala.Math.ceil`

Reply:
Addressed